### ISSUE TITLE
Unsupported query: unknown signature: extract(string, interval)

### ISSUE THREAD

**Issue by timgraham** [cockroachdb/django-cockroachdb#29]
When using `extract()` on a `DurationField`:
```
======================================================================
ERROR: test_extract_duration (db_functions.datetime.test_extract_trunc.DateFunctionTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tim/code/django/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.UndefinedFunction: unknown signature: extract(string, interval)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/tim/code/django/django/test/testcases.py", line 1229, in skip_wrapper
    return test_func(*args, **kwargs)
  File "/home/tim/code/django/tests/db_functions/datetime/test_extract_trunc.py", line 242, in test_extract_duration
    lambda m: (m.start_datetime, m.extracted)
  File "/home/tim/code/django/django/test/testcases.py", line 1047, in assertQuerysetEqual
    items = map(transform, qs)
  File "/home/tim/code/django/django/db/models/query.py", line 274, in __iter__
    self._fetch_all()
  File "/home/tim/code/django/django/db/models/query.py", line 1242, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/home/tim/code/django/django/db/models/query.py", line 55, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/home/tim/code/django/django/db/models/sql/compiler.py", line 1100, in execute_sql
    cursor.execute(sql, params)
  File "/home/tim/code/django/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/home/tim/code/django/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/home/tim/code/django/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/home/tim/code/django/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/tim/code/django/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.ProgrammingError: unknown signature: extract(string, interval)
```
Is this a missing feature of cockroachdb or is it required (correct?) to use the `extract_duration()` function instead of `extract()`?

Sample SQL:
```sql
SELECT ... EXTRACT('second' FROM "db_functions_dtmodel"."duration") AS "extracted"
FROM "db_functions_dtmodel" 
```

**Comment by jordanlewis** [comment]
This is a missing feature that was just fixed but won't be backported to 19.2. The workaround is to use the SQL standard extract which uses keywords instead of strings. So it'd be `EXTRACT(SECOND FROM ...)`.

**Comment by timgraham** [comment]
That's actually the form the cockroachdb adapter is using. (I pulled the sample SQL, with quotes around 'second', from PostgreSQL.) I already [overrode the PostgreSQL adapter](https://github.com/cockroachdb/cockroach-django/blob/7795739383372bdffe76d55af194271c06fc4159/cockroach/django/operations.py#L42) to omit the quotes around the `lookup_type`.

**Comment by timgraham** [comment]
Does cockroachdb support `extract()` on an `interval` data type?

**Comment by timgraham** [comment]
Fixed in CockroachDB 20.1: https://github.com/cockroachdb/cockroach/issues/43209.
